pkg/apis/pipeline/v1beta1/sourceversion_types.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SourceVersionSpec defines the desired state of SourceVersion
+type SourceVersionSpec struct {
+	SourceRef SourceRef `json:"sourceRef"`
+	Version   string    `json:"string"`
+}
+
+// SourceVersionStatus defines the observed state of TaskRun
+type SourceVersionStatus struct {
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SourceVersion is the Schema for the sourceversion API
```

Comment:
maybe a bit here about how a `SourceVersion` would be intended to be used?

Reply:
SourceVersion is for the `taskrun_controller` to make sure the same version of the Source (eg: git commit) is used across the same pipeline run, will update the comment